packages/components/date-picker/src/date-picker-com/basic-date-table.vue

```diff
@@ -42,407 +42,407 @@
-          } else {
-            if (count <= dateCountOfMonth) {
-              cell.text = count++
-            } else {
-              cell.text = count++ - dateCountOfMonth
-              cell.type = 'next-month'
-            }
-          }
-
-          const cellDate = calTime.toDate()
-          cell.selected = selectedDate.find(
-            (_) => _.valueOf() === calTime.valueOf()
-          )
-          cell.isSelected = !!cell.selected
-          cell.isCurrent = isCurrent(cell)
-          cell.disabled = props.disabledDate && props.disabledDate(cellDate)
-          cell.customClass =
-            props.cellClassName && props.cellClassName(cellDate)
-          row[props.showWeekNumber ? j + 1 : j] = cell
-        }
+const tbodyRef = ref<HTMLElement>()
+const currentCellRef = ref<HTMLElement>()
+// data
+const lastRow = ref<number>()
+const lastColumn = ref<number>()
+const tableRows = ref<DateCell[][]>([[], [], [], [], [], []])
 
-        if (props.selectionMode === 'week') {
-          const start = props.showWeekNumber ? 1 : 0
-          const end = props.showWeekNumber ? 7 : 6
-          const isActive = isWeekActive(row[start + 1])
-          row[start].inRange = isActive
-          row[start].start = isActive
-          row[end].inRange = isActive
-          row[end].end = isActive
-        }
-      }
-      return rows_
-    })
+// todo better way to get Day.js locale object
+const firstDayOfWeek = (props.date as any).$locale().weekStart || 7
+const WEEKS_CONSTANT = props.date
+  .locale('en')
+  .localeData()
+  .weekdaysShort()
+  .map((_) => _.toLowerCase())
 
-    watch(
-      () => props.date,
-      async () => {
-        if (tbodyRef.value?.contains(document.activeElement)) {
-          await nextTick()
-          currentCellRef.value?.focus()
-        }
-      }
-    )
+const offsetDay = computed(() => {
+  // Sunday 7(0), cal the left and right offset days, 3217654, such as Monday is -1, the is to adjust the position of the first two rows of dates
+  return firstDayOfWeek > 3 ? 7 - firstDayOfWeek : -firstDayOfWeek
+})
 
-    const focus = async () => {
-      currentCellRef.value?.focus()
-    }
+const startDate = computed(() => {
+  const startDayOfMonth = props.date.startOf('month')
+  return startDayOfMonth.subtract(startDayOfMonth.day() || 7, 'day')
+})
 
-    const isCurrent = (cell): boolean => {
-      return (
-        props.selectionMode === 'date' &&
-        (cell.type === 'normal' || cell.type === 'today') &&
-        cellMatchesDate(cell, props.parsedValue)
-      )
-    }
+const WEEKS = computed(() => {
+  return WEEKS_CONSTANT.concat(WEEKS_CONSTANT).slice(
+    firstDayOfWeek,
+    firstDayOfWeek + 7
+  )
+})
 
-    const cellMatchesDate = (cell, date) => {
-      if (!date) return false
-      return dayjs(date)
-        .locale(lang.value)
-        .isSame(props.date.date(Number(cell.text)), 'day')
-    }
+const hasCurrent = computed<boolean>(() => {
+  return rows.value.flat().some((row) => {
```

Comment:
`Array.prototype.flat` is not supported in Chrome 64.
https://caniuse.com/array-flat

Reply:
I think we can add polyfill from core-js.